Fix application of optimized iterator for group by queries with a single key (#15134)
crate/crate: docs/appendices/release-notes/5.5.2.rst

Fixes
=====

> - Fixed a performance regression introduced in 5.5.0, ensuring that the
> optimization of group by queries with a single column is applied.

Reviewer: ```suggestion
- Fixed a performance regression introduced in 5.5.0 for queries with
   ``GROUP BY`` on a single column.
```
